### PR TITLE
Fixes #12865: CommandView Cursor Icons use hardcoded XPM

### LINKED ISSUE #12865: View: CommandView Cursor Icons use hardcoded XPM
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Problem description

There are still hardcoded XPM cursor icons in the CommandView.cpp and they appear too large on screen and pixelated on HiDPI monitors. They should be replaced with a .svg icon:
- Box Zoom https://github.com/FreeCAD/FreeCAD/blob/addf08ce6b0df264a47e836e799b6756626f146e/src/Gui/CommandView.cpp#L2071
grafik
- ViewBoxZoom https://github.com/FreeCAD/FreeCAD/blob/addf08ce6b0df264a47e836e799b6756626f146e/src/Gui/CommandView.cpp#L2676
- Box Selection https://github.com/FreeCAD/FreeCAD/blob/addf08ce6b0df264a47e836e799b6756626f146e/src/Gui/CommandView.cpp#L2747
- Box Element Select https://github.com/FreeCAD/FreeCAD/blob/addf08ce6b0df264a47e836e799b6756626f146e/src/Gui/CommandView.cpp#L3005
grafik
- Measure Distance https://github.com/FreeCAD/FreeCAD/blob/addf08ce6b0df264a47e836e799b6756626f146e/src/Gui/CommandView.cpp#L3229
grafik

### Full version info

```shell
OS: Windows 11 build 22631
Word size of FreeCAD: 64-bit
Version: 0.22.0dev.36277 (Git)
Build type: Release
Branch: main
Hash: 9e1903d46112b3660bf10c6a4537d728101d560b
Python 3.10.13, Qt 5.15.8, Coin 4.0.2, Vtk 9.2.6, OCC 7.6.3
Locale: German/Germany (de_DE)
Installed mods: 
  * 3DfindIT 1.2.0
  * BIM 2021.12.0
  * CfdOF 1.25.2
  * CurvedShapes 1.0.5
  * Curves 0.6.23
  * Defeaturing 1.2.2
  * fasteners 0.5.12
  * FEMbyGEN 2.1.0
  * freecad.gears 1.2.0
  * freecad_metal_workbench 0.0.1
  * OpenDark 2023.12.17
  * sheetmetal 0.4.2
```

### Subproject(s) affected?

Core

### Anything else?

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by wwmayer**

### CODE REVIEW

File: src/Gui/CommandView.cpp [hunk 1]
```diff
@@ -2767,7 +2729,8 @@ void StdViewBoxZoom::activated(int iMsg)
     if ( view ) {
         View3DInventorViewer* viewer = view->getViewer();
         if (!viewer->isSelecting()) {
-            SelectionCallbackHandler::Create(viewer, View3DInventorViewer::BoxZoom, QCursor(QPixmap(cursor_box_zoom), 7, 7));
+            QPixmap px(Gui::BitmapFactory().pixmapFromSvg("zoom-border-cross", QSize(32, 32)));
```
Comment (maxwxyz) [inline src/Gui/CommandView.cpp:2732]: it looks pixelated on HiDPI screens.
Would it work with: `32 * devicePixelRatio()` and also for the `QCursor(px, 6 * devicePixelRatio(), 6 * devicePixelRatio())` ?

**Review by maxwxyz — CHANGES_REQUESTED** [review]
Just a question if this would solve the pixelation with size * devicePixelRatio()

Is there a global option to pull the pixelRatio like `getViewer()->devicePixelRatio()` ?

**Comment by wwmayer** [comment]
> Is there a global option to pull the pixelRatio like getViewer()->devicePixelRatio()

It's recommended to get the DPR from the widget you are working on.

File: src/Gui/CommandView.cpp (hunk 1 above)
Comment (wwmayer) [reply src/Gui/CommandView.cpp:2732]: What's probably needed is to pass the size of 32 * devicePixelRatio() and then set the device pixel ratio to the returned pixmap. The hotspot of the cursor only needs to be adjusted for some Linux platforms while on macOS and Windows it's done automatically.

That's the way how it's done e.g. in DrawSketchHandler::setSvgCursor

**Comment by maxwxyz** [comment]
@wwmayer will you change it in this PR?

**Comment by wwmayer** [comment]
Yes, but later...

**Review by maxwxyz — APPROVED** [review]
nice, thank you!
Could I use some of the code for tree overlays which have the same problem (pixelation) in PR https://github.com/FreeCAD/FreeCAD/pull/13110

Unfortunately there I  don't have hotX/Y values. Maybe you could help me there.

**Comment by wwmayer** [comment]
> nice, thank you!

Does it mean that all issues are solved?

> Unfortunately there I don't have hotX/Y values. Maybe you could help me there.

All what you can reuse is the part to scale up the size and set the device pixel ratio to the pixmap. But as said an overloaded method can be added to the BitmapFactoryInst class to do all the magic. Then in the TreeView class it can be called this way: `pxError = Gui::BitmapFactory().pixmapFromSvg("overlay_error", QSize(10, 10), treeWidget()->devicePixelRatioF());`

**Comment by wwmayer** [comment]
https://github.com/FreeCAD/FreeCAD/pull/13176

**Pull request merged**